tests/cpu_bugs/xen_mitigations.pm

```diff
@@ -248,47 +248,75 @@ my $mitigations_list =
         "l1d-flush=off" => {
             default => {
                 expected   => {''},
-                unexpected => {'xl dmesg' => ['Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other: .*L1D_FLUSH']},
+                unexpected => {'xl dmesg' => ['Xen settings: BTI-Thunk .*, SPEC_CTRL: IBRS. SSBD- .*, Other: .*L1D_FLUSH']},
             }
         },
         "l1d-flush=on" => {
             default => {
-                expected   => {'xl dmesg' => ['Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other: .*L1D_FLUSH']},
+                expected   => {'xl dmesg' => ['Xen settings: BTI-Thunk .*, SPEC_CTRL: IBRS. SSBD- .*, Other: .*L1D_FLUSH']},
                 unexpected => {''},
             }
         },
+        "branch-harden=on" => {
+            default => {
+                expected   => {'xl dmesg' => ['Xen settings: BTI-Thunk .*, SPEC_CTRL: IBRS. SSBD- .*, Other: .*BRANCH_HARDEN']},
+                unexpected => {''},
+            }
+        },
+        "branch-harden=off" => {
+            default => {
+                expected   => {''},
+                unexpected => {'xl dmesg' => ['Xen settings: BTI-Thunk .*, SPEC_CTRL: IBRS. SSBD- .*, Other: .*BRANCH_HARDEN']},
+            }
+        },
     },
   };
 
+sub check_expected_string {
```

Comment:
Maybe combine this two functions into one?
check_string  with a parameter true or false.